### PR TITLE
better deconversion

### LINKED ISSUE #21828: revs dont lose briefing when headrev dies
## Description
maybe role as well, and maybe when mindshielded too not sure

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
when all headrevs die:
- all revs get a window explaining they got deconverted
- role is now removed when deconverted

## Why / Balance
- during a fight its hard to see the popup or losing icons, a window explaining that is imposible to ignore.
- fixes #21828

## Technical details
first xaml eui real??

## Media
the window

no longer valid

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- tweak: Revolutionaries that are deconverted by headrevs dying get a window shown explaining that they are no longer revs, to help with tunnel vision.

### CODE REVIEW

**Comment by Rainbeon** [comment]
Is "You forget everything that happened since being converted" even true though? It does not seem to be part of the rules nor MRP's New Life Rule. Seems like a weird inclusion.

**Comment by deltanedas** [comment]
its not a rule but:
1. makes it clearer that revs arent liable for anything they did :trollface: which is already how its done in practice
2. you shouldnt really act on information while you are hypnotised

this is based on gameplay from mrp idk how lrp shitters handle it

**Comment by deltanedas** [comment]
could remove the amnesia bit if admins think its best

**Comment by Rainbeon** [comment]
If we want a rule change on how Deconversion memory works that should come separately from this no? That's an over-arching rule change and the only way to learn about it is through this specific end-game prompt (not even on mind-shielding deconversion, which is where this seems more important to mention...).

>revs cant testify to what a headrev did while they are converted

This is a *very* large change in the dynamics of a Revs round than what seems relevant to the scope of this PR (notification of deconversion), and seems far more important to mention during a mind-shielding than in the aftermath of HRs dying anyway, as all it really does when being mentioned there is remove any RP relating to the consequences of your actions/coping and such.

I'd rather see the amnesia mention removed and memory retention decided upon as an actual rule change from admins first, given how much it changes.

**Comment by GreyMario** [comment]
Consensus is that Revolutionary conversion is brainwashing, not hypnosis. You are fully aware and have memory.

**Comment by JerryImMouse** [comment]
Is it a good idea to make UI with C#? I think it can be done with XAML easier and will make the code more readable

File: Content.Client/Revolutionary/UI/DeconvertedMenu.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                                    Text = Loc.GetString("rev-deconverted-confirm")
+                                })
+                            }
+                        },
+                    }
+                },
+            }
+        });
+    }
+}
```
Comment (metalgearsloth) [inline Content.Client/Revolutionary/UI/DeconvertedMenu.cs:47]: XAML please.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Client/Revolutionary/UI/DeconvertedMenu.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Client/Revolutionary/UI/DeconvertedMenu.cs:47]: is there an example of a xaml eui i can steal :trollface:

Comment (deltanedas) [reply Content.Client/Revolutionary/UI/DeconvertedMenu.cs:47]: @metalgearsloth a

**Comment by ps3moira** [comment]
thank you 🙏  unconverted revs gibbing clown will now know they're not rev and gib clown in peace.

File: Content.Client/Revolutionary/UI/DeconvertedMenu.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Client/Revolutionary/UI/DeconvertedMenu.cs:47]: ok probably done right

Comment (metalgearsloth) [reply Content.Client/Revolutionary/UI/DeconvertedMenu.cs]: you think I read my github pings noob

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**